modules/imgcodecs/src/ios_conversions.mm

```diff
@@ -116,6 +118,8 @@ void UIImageToMat(const UIImage* image,
         if (!alphaExist)
             bitmapInfo = kCGImageAlphaNoneSkipLast |
                                 kCGBitmapByteOrderDefault;
+        else
+            m = cv::Scalar(0);
```

Comment:
it's the same cost - whether you set the whole image to 0's or just alpha channel, since the data is interleaved. I would say, in principle, setting the whole image to 0's may be even faster, if compiler generates proper instructions for it.